1.21.5 update

Big-Iron-Cheems opened the pull request:
## Type of change

- [ ] Bug fix
- [x] New feature

## Description

1.21.5 update.

## Related issues

None.

# How Has This Been Tested?

Not yet tested.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have added comments to my code in more complex areas.
- [ ] I have tested the code in both development and production environments.

Big-Iron-Cheems committed:
Initial commit

TODO: update fabric-renderer-indigo version when it releases

Wide-Cat committed:
fix a bunch of errors

Wide-Cat committed:
Fix even more errors

Issues with optionals should hopefully now be resolved, unless I missed any. Major issues remaining are with rendering changes, removal of tool/weapon items, and inventory changes.

Big-Iron-Cheems committed:
Mixins + version bump

Wide-Cat committed:
Fix issues with inventory and tool/weapon changes

Big-Iron-Cheems committed:
Dependency bump

Fabric renderer Indigo was updated

Powie69 commented:
yeahhhh margee itt!!!!!!!

Powie69 commented:
merge*

Big-Iron-Cheems committed:
Dependency bump 1.21.5 stable

Big-Iron-Cheems committed:
Dependency bump

MineGame159 committed:
Fixed rendering + some other things

MineGame159 committed:
Fix scissor not being applied to items in GUI

MineGame159 committed:
Fixed line rendering in Renderer2D

crosby-moe committed:
fix EntityOwner npe

crosby-moe committed:
fix meteor toast

crosby-moe committed:
`RegistryListSettingScreen` -> `CollectionListSettingScreen`
relax parameter from `Registry<T>` to `Iterable<T>`

crosby-moe committed:
remove `PacketUtils.PacketRegistry` in favor of a set

Big-Iron-Cheems committed:
Merge pull request #6 from RacoonDog/1.21.5-update-pr

1.21.5 update pr

Big-Iron-Cheems committed:
Dependency bump

Also suppress some deprecation warns

Big-Iron-Cheems committed:
Properly suppress warns

Added compile args to provide better logs in the future, suppress launch subproject warns.

Big-Iron-Cheems committed:
PacketUtilsUtil fix filter

Previous comparison wouldn't skip the excluded packets

Wide-Cat committed:
Merge branch 'master' into 1.21.5-update

Big-Iron-Cheems committed:
Fix merge commit

Susmaster64 commented:
Every module works (Turns on and off properly) when tested.

Big-Iron-Cheems committed:
Minor cleanup

RedCarlos26 commented:
Anything else to fix? (Excluding freecam?

Susmaster64 commented:
Is it just the running the locally built jar in prismlauncher? Or does the artifact from github actions also have the same problem?
Builds and runs in prism fine on my end.

Wide-Cat commented:
> Show hands not work in Freecam ![изображение](https://private-user-images.githubusercontent.com/35975332/429466475-7df2ebdc-59a3-41df-a8de-6a712bd6e57d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3Njc1OTgsIm5iZiI6MTc0Mzc2NzI5OCwicGF0aCI6Ii8zNTk3NTMzMi80Mjk0NjY0NzUtN2RmMmViZGMtNTlhMy00MWRmLWE4ZGUtNmE3MTJiZDZlNTdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDExNDgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxM2FiNjg2MmMyNTljYjRhM2ExNTJlMWM2OTA1NDQyZWZkMTFmOTUxNmQ2NjY5ZWJjMThlYzY2ODI3YjY3Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MY-JDmKdrM3Qw29zQOb7n3tyRgAUpaqKGoxXayYugV0)

Unable to reproduce. Unless Big Iron fixed it in his changes, which it doesn't look like happened, maybe it's a mod/addon/module conflict problem.

![java_CmRhLBxqjn](https://github.com/user-attachments/assets/b953ef2a-ecad-4dda-a30c-9b83dad9eac4)

MineGame159 committed:
Include fabric-api-base in the built jar, since it is required by fabric-resource-loader-v0 now

Big-Iron-Cheems committed:
Handle planned deprecation

Add missing Suppress too

crosby-moe committed:
remove module registry

crosby-moe committed:
deprecate `Modules#getList`

crosby-moe committed:
cache snbt reader

Big-Iron-Cheems committed:
Merge pull request #7 from RacoonDog/module-deprecations

Module deprecations

Big-Iron-Cheems committed:
Merge pull request #8 from RacoonDog/cache-snbt-reader

cache snbt reader

Nekiplay commented:
> > Show hands not work in Freecam ![изображение](https://private-user-images.githubusercontent.com/35975332/429466475-7df2ebdc-59a3-41df-a8de-6a712bd6e57d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3Njc1OTgsIm5iZiI6MTc0Mzc2NzI5OCwicGF0aCI6Ii8zNTk3NTMzMi80Mjk0NjY0NzUtN2RmMmViZGMtNTlhMy00MWRmLWE4ZGUtNmE3MTJiZDZlNTdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDExNDgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxM2FiNjg2MmMyNTljYjRhM2ExNTJlMWM2OTA1NDQyZWZkMTFmOTUxNmQ2NjY5ZWJjMThlYzY2ODI3YjY3Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MY-JDmKdrM3Qw29zQOb7n3tyRgAUpaqKGoxXayYugV0)
> 
> Unable to reproduce. Unless Big Iron fixed it in his changes, which it doesn't look like happened, maybe it's a mod/addon/module conflict problem.
> 
> ![java_CmRhLBxqjn](https://private-user-images.githubusercontent.com/74723656/430365960-b953ef2a-ecad-4dda-a30c-9b83dad9eac4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM4MTEyNDksIm5iZiI6MTc0MzgxMDk0OSwicGF0aCI6Ii83NDcyMzY1Ni80MzAzNjU5NjAtYjk1M2VmMmEtZWNhZC00ZGRhLWEzMGMtOWI4M2RhZDllYWM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDIzNTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBlZTY5ZTdhNGVkNzM0M2ZkYmNjMDcxYzNkZWQ0NWY5NzExMjU3YTlmOTM5MGE5MTg3NGRmNWYyZGRhNWZkOWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GXHl7-JefKw3TAd4ZbcXs8F3ee2ZyoygzoTi1l-EMNs)

Fixed in last update

Big-Iron-Cheems committed:
PotionSpoof rework

Spoofing clientside effects is unreliable/broken on pretty much any server.
Module rebranded as NoStatusEffects with only effect blocking capability.

19MisterX98 commented:
Is there anything that is still broken?

OmarRpp commented:
nice

Wide-Cat committed:
Merge branch 'master' into 1.21.5-update

Wide-Cat committed:
fixes and improvements

Wide-Cat commented:
Known issues:
- Shader ESP is unusably broken
- Strafe speed accumulates speed in the air nonstop while holding forward/backward and left/right at the same time

19MisterX98 commented:
I noticed the shader esp issue too yesterday after testing the pr a bit.
Chams has the same issue and i narrowed it down to this phenomenon.
Seems like entities with multiple things to render break it.

https://github.com/user-attachments/assets/0f7eacc0-107f-4c2b-b9ae-c1c020f12afa

RedCarlos26 commented:
> I noticed the shader esp issue too yesterday after testing the pr a bit.
> 
> Chams has the same issue and i narrowed it down to this phenomenon.
> 
> Seems like entities with multiple things to render break it.
> 
> 
> 
> https://github.com/user-attachments/assets/0f7eacc0-107f-4c2b-b9ae-c1c020f12afa
> 
> 
> 
> 

Are you using iris? It happens to me even in 1.21.4

MineGame159 committed:
Fix Shader ESP and Chams
